### ISSUE TITLE
Unable to get storage settings

### ISSUE THREAD

**Issue by imobachgs** [openSUSE/agama#1624]
Using the latest versions as of today:

* `agama-cli-9+969-48.1.x86_64`
* `agama-products-opensuse-9+881-14.1.noarch`
* `agama-web-ui-9+969-68.1.noarch`
* `agama-integration-tests-9+953-10.1.noarch`
* `agama-auto-9+953-12.1.noarch`
* `ruby3.3-rubygem-agama-yast-9.devel964-34.1.x86_64`
* `agama-9+969-48.1.x86_64`

The storage service is unable to return the settings, so visiting the storage page results in a blank page. According to the browser, the request to `/api/storage/proposal/settings` returns in a `400 Bad Request` error. The problem is visible at D-Bus level:

```
‣ Type=method_call  Endian=l  Flags=0  Version=1 Cookie=82  Timestamp="Fri 2024-09-20 16:04:28.142402 UTC" 
  Sender=:1.1  Destination=org.opensuse.Agama.Storage1  Path=/org/opensuse/Agama/Storage1/Proposal  Interface=org.freedesktop.DBus.Properties  Member=Get 
  UniqueName=:1.1 
  MESSAGE "ss" { 
          STRING "org.opensuse.Agama.Storage1.Proposal"; 
          STRING "Actions"; 
  }; 
 
‣ Type=error  Endian=l  Flags=0  Version=1 Cookie=137  ReplyCookie=81  Timestamp="Fri 2024-09-20 16:04:28.200765 UTC" 
  Sender=:1.2  Destination=:1.1 
  ErrorName=org.freedesktop.DBus.Error.Failed  ErrorMessage="undefined method `size' for nil; caused by 1 sender=:1.1 -> dest=org.opensuse.Agama.Storage1 serial=81 reply_serial= path=/org/opensuse/Agama/Storage1/Proposal; interface=org.freedesktop.DBus.Properties; member=Get error_name=" 
  UniqueName=:1.2 
  MESSAGE "sas" { 
          STRING "undefined method `size' for nil; caused by 1 sender=:1.1 -> dest=org.opensuse.Agama.Storage1 serial=81 reply_serial= path=/org/opensuse/Agama/Storage1/Proposal; interface=org.freedesktop.DBus.Properties; member=Get error_name="; 
```

I am attaching the logs: [agama-logs.tar.gz](https://github.com/user-attachments/files/17077750/agama-logs.tar.gz)

**Comment by imobachgs** [comment]
I am not an expert in the storage area, but it might be a problem when trying to convert to D-Bus. The call to [ProposalSettingsConversion.to_dbus](https://github.com/openSUSE/agama/blob/master/service/lib/agama/dbus/storage/proposal.rb#L57) returns this:

```
{"Target"=>"disk", "TargetDevice"=>"/dev/vda", "ConfigureBoot"=>true, "BootDevice"=>"", "DefaultBootDevice"=>"/dev/vda", "EncryptionPassword"=>"", "EncryptionMethod"=>"luks2", "EncryptionPBKDFunction"=>"pbkdf2", "SpacePolicy"=>"delete", "SpaceActions"=>[{"Device"=>#<Y2Storage::SpaceActions::Delete:0x00007ffa5fbcbdd8 @device="/dev/vda1", @mandatory=true>, "Action"=>""}, {"Device"=>#<Y2Storage::SpaceActions::Delete:0x00007ffa5fbcbd60 @device="/dev/vda2", @mandatory=true>, "Action"=>""}, {"Device"=>#<Y2Storage::SpaceActions::Delete:0x00007ffa5fbcbd38 @device="/dev/vda3", @mandatory=true>, "Action"=>""}], "Volumes"=>[{"MountPath"=>"/", "MountOptions"=>[], "Target"=>"default", "TargetDevice"=>"", "FsType"=>"Btrfs", "MinSize"=>18790481920, "AutoSize"=>true, "Snapshots"=>true, "Transactional"=>false, "Outline"=>{"Required"=>true, "FsTypes"=>["Btrfs", "Ext2", "Ext3", "Ext4", "XFS"], "SupportAutoSize"=>true, "AdjustByRam"=>false, "SnapshotsConfigurable"=>true, "SnapshotsAffectSizes"=>true, "SizeRelevantVolumes"=>["/home"]}}, {"MountPath"=>"swap", "MountOptions"=>[], "Target"=>"default", "TargetDevice"=>"", "FsType"=>"Swap", "MinSize"=>4108935168, "AutoSize"=>true, "Snapshots"=>false, "Transactional"=>false, "Outline"=>{"Required"=>false, "FsTypes"=>["Swap"], "SupportAutoSize"=>true, "AdjustByRam"=>true, "SnapshotsConfigurable"=>false, "SnapshotsAffectSizes"=>false, "SizeRelevantVolumes"=>[]}, "MaxSize"=>4108935168}]}
```

To be honest, the space actions look wrong to me:

```
{"Device"=>#<Y2Storage::SpaceActions::Delete:0x00007ffa5fbcbdd8 @device="/dev/vda1", @mandatory=true>, "Action"=>""}
```

Compared to an older (and working) version:

```
{"Device"=>"/dev/vda3", "Action"=>"force_delete"}
```

**Comment by imobachgs** [comment]
Fixed in #1628.
